cwms_radar_api/src/main/java/cwms/radar/api/LevelsController.java

```diff
@@ -193,74 +187,150 @@ public void delete(@NotNull Context ctx, String id) {
+                    beginZdt = endZdt.minusHours(24);
                 }
-                case "png": // fall next
-                case "jpg": // fall next
-                default: {
-                    ctx.status(HttpServletResponse.SC_NOT_IMPLEMENTED)
-                            .json(RadarError.notImplemented());
+
+                LocationLevels levels = levelsDao.getLocationLevels(cursor, pageSize, names,
+                        office, unit, datum, beginZdt, endZdt);
+                String result = Formats.format(contentType, levels);
+
+                ctx.result(result).contentType(contentType.toString());
+                requestResultSize.update(result.length());
+
+                ctx.status(HttpServletResponse.SC_OK);
+            } else {
+
+
+                switch (format) {
+                    case "json": {
+                        ctx.contentType(Formats.JSON);
+                        break;
+                    }
+                    case "tab": {
+                        ctx.contentType(Formats.TAB);
+                        break;
+                    }
+                    case "csv": {
+                        ctx.contentType(Formats.CSV);
+                        break;
+                    }
+                    case "xml": {
+                        ctx.contentType(Formats.XML);
+                        break;
+                    }
+                    case "wml2": {
+                        ctx.contentType(Formats.WML2);
+                        break;
+                    }
+                    case "png": // fall next
+                    case "jpg": // fall next
+                    default: {
+                        ctx.status(HttpServletResponse.SC_NOT_IMPLEMENTED).json(RadarError.notImplemented());
+                    }
                 }
-            }
 
-            String results = levelsDao.getLocationLevels(format, names, office, unit, datum,
-                    begin, end, timezone);
-            ctx.status(HttpServletResponse.SC_OK);
-            ctx.result(results);
-            requestResultSize.update(results.length());
+                String results = levelsDao.getLocationLevels(format, names, office, unit, datum,
+                        begin, end, timezone);
+                ctx.status(HttpServletResponse.SC_OK);
+                ctx.result(results);
+                requestResultSize.update(results.length());
+            }
         }
     }
 
 
-    @OpenApi(ignore = true)
+    @OpenApi(
+            pathParams = {
+                    @OpenApiParam(name = "id", required = true, description = "Specifies"
+                            + " the requested location level."),
+            },
+            queryParams = {
+                    @OpenApiParam(name = "office", required = true, description = "Specifies the "
+                            + "office of the Location Level to be returned"),
+                    @OpenApiParam(name = "date", required = true, description = "Specifies the "
```

Comment:
should this be "date-time" since it can go down the 2nd?

Reply:
Maybe?  I didn't think too hard about it b/c the update and delete already had a "date" field.  But maybe all three should be "date-time" or something more explicit like "effective-date" ?

Reply:
I like effective-date more. that's explicitly what we're getting/removing/or updating.